### ISSUE TITLE
Make allocation failure error messages more readable.

### ISSUE THREAD

**Issue by sangeethah** [rancher/rancher#8653]
Rancher Versions: v1.6.0-rc4

Steps to reproduce the problem:
Create a service with ebs volume.
Put the host where the service container is running in "inactive" state.
Try to scale up the service.
New service instances will fail to start (which is as expected) , with the following error messages:

```
Degraded (Failed to allocate instance [container:1i182]: Bad instance [container:1i182] in state [error]: Allocation failed: Volume 248 can only be used on host 5)
```

This error message can be made more readable with having host/volume name instead of their ids.

**Comment by sangeethah** [comment]
Tested with rancher-server build from v1.6 branch. 

Following is the error message seen:

```Degraded (Failed to allocate instance [container:1i60]: Bad instance [container:1i60] in state [error]: Allocation failed: Volume mytest1_testvol1_de5ed(130) can only be used on host sanebstest1(3) and resource constraints: [1 instanceReservation])```

This resource constraints error message is not readable . Also instance/host/volume ids are still included in the error message.

**Comment by cjellick** [comment]
A few thoughts:
1. We're need keep the IDs because they are very valuable for debugging, but they need to be obfuscated (should read 1v130 and 1h3) and we should mention that they are ids
2. The grammar of "and resource constraints" is not very good. Maybe we can improve by making it a standalone sentence and only including it when necessary.
3. We should just omit "instanceReservation from the resource constraints and don't include the resource constraints clause if it is empty.

So, the message should look like:
```
Degraded (Failed to allocate instance [container:1i60]: Bad instance [container:1i60] in state [error]: Allocation failed: Volume mytest1_testvol1_de5ed (id: 1v130) can only be used on host sanebstest1 (id: 1h3).
```

If a real resourceConstaint (like memoryReservation) is applied, it should look like:
```
Degraded (Failed to allocate instance [container:1i60]: Bad instance [container:1i60] in state [error]: Allocation failed: Host must satisfy resource constraints: [memoryReservation: 1000]. Volume mytest1_testvol1_de5ed (id: 1v130) can only be used on host sanebstest1 (id: 1h3). 
```

What do you think of those @sangeethah? (We're keeping the ids, they're too important for debugging, that's not debatable)

**Comment by sangeethah** [comment]
@cjellick I am fine with your suggestions.
@StrongMonkey Reopening this issue for addressing these changes.

**Comment by StrongMonkey** [comment]
Thanks for the explanation @sangeethah @cjellick. I ll make the change

**Comment by StrongMonkey** [comment]
Available at master now.

**Comment by sangeethah** [comment]
Tested with v1.63-rc2

Volume and Host Ids are still not obfuscated.

Following is the error message:

 ```Degraded (Failed to allocate instance [container:1i80]: Bad instance [container:1i80] in state [error]: Allocation failed: Volume hello123_testvol1_b2915(id: 133) can only be used on host soumyaawshostadd2(id: 13))```

**Comment by sangeethah** [comment]
@StrongMonkey @cjellick Is this going to be addressed for 1.6.3. I see partial fixes  (like "id" being included)?

**Comment by sangeethah** [comment]
Tested with rancher-server version - v1.6.3-rc4.
volume and host ids are obfuscated in error message as expected.
Following is the error message seen:

```Degraded (Failed to allocate instance [container:1i23]: Bad instance [container:1i23] in state [error]: Allocation failed: Volume testme_testvol1_ef902(id: 1v50) can only be used on host sanaws1(id: 1h3))```
